internal/controller/pvc_builder.go

```diff
@@ -178,10 +168,11 @@ func (pb *PVCBuilder) UpdatePVCSpec(ctx context.Context, existingPVC *corev1.Per
 		return fmt.Errorf("cannot update PVC to nil spec")
 	}
 
-	// Update only the fields we control (preserve Kubernetes-managed fields)
+	// Update only the MUTABLE fields (preserve immutable and Kubernetes-managed fields)
+	// Note: StorageClassName is immutable after creation and cannot be updated
 	existingPVC.Spec.AccessModes = desiredPVC.Spec.AccessModes
 	existingPVC.Spec.Resources = desiredPVC.Spec.Resources
-	existingPVC.Spec.StorageClassName = desiredPVC.Spec.StorageClassName
+	// DO NOT update StorageClassName - it's immutable after PVC creation
```

Comment:
just to confirm: this change comes from another branch? otherwise curious why you are making this change (which seems sound btw)